Documentation/tutorial/intro_template.md

```diff
@@ -0,0 +1,170 @@
+Introduction to *DocFX Template System*
+===============================
+
+*DocFX Template System* provides a flexible way of defining and using templates.
+
+As the following DocFX workflow shows,
+
+![DocFX workflow](../spec/images/docfx_workflow_highlevel.png)
+
```

Comment:
It's hard to tell the difference between between `Document Processors`/`Preprocessors`/`Renderers` at first glance. Add some notation at the column's bottom? e.g.:
Document Processors: build-in plugin
Preprocessors: '.js' script
Renderers: '.tmpl' in Mustach or '.liquid' inLiquid